File: LeetCode/Kai/62UniquePaths.java
public class Solution {
    public int uniquePaths(int m, int n) {
        if(m==1 && n==1) return 1;  //corner case
        
        int[][] dp = new int[m][n];
        //base
        for(int i=1; i<m; i++) {
            dp[i][0] = 1;
        }
        for(int i=1; i<n; i++) {
            dp[0][i] = 1;
        }
        
        for(int i=1; i<m; i++) {
            for(int j=1; j<n; j++) {
                dp[i][j] = dp[i-1][j]+dp[i][j-1];
            }
        }
        return dp[m-1][n-1];
    }
    
}

File: LeetCode/Liang/62UniquePaths.cpp
/**
 * Dp
 **/

class Solution {
public:
    int uniquePaths(int m, int n) {
        assert(m > 0 && n > 0);
        vector<int> DP(n, 1);
        for (int i = 1; i < m; ++i) {
            for (int j = 1; j < n; ++j) {
                DP.at(j) = DP.at(j) + DP.at(j - 1);
            }
        }
        return DP.back();
    }
};